src/dev/flang/fuir/FUIR.java

```diff
@@ -2337,45 +2295,37 @@ public int codeIndex(int c, int ix, int delta)
    * NYI: Performance: This requires time 'O(codeSize(c))', so using this
    * quickly results in quadratic performance!
    *
-   * @param c the code block
-   *
-   * @param i current index in c, starting at 0.
+   * @param si a site, our current position we are checking
    *
-   * @param ix an index in c
+   * @param s a site we are looking for
    *
    * @param delta the negative number of instructions to go back.
    *
-   * @return the index of the expression 'n' expressions before 'ix', or a
+   * @return the site of the expression 'delta' expressions before 's', or a
    * negative value '-m' if that instruction can be found 'm' recursive calls up.
    */
-  private int codeIndex2(int c, int i, int ix, int delta)
+  private int codeIndex2(int si, int s, int delta)
   {
-    if (i == ix)
```

Comment:
What is the meaning of this?

Reply:
The meaning is that `c`/`i` is at the site we are looking for. This has been replaced by `si == s` a few lines below, where I added a corresponding comment to the `return` since I found the original code too confusion.

Reply:
I read this the wrong way. I thought the line with if is still present, but it has been removed...